filters/include/pcl/filters/filter_indices.h

```diff
@@ -302,6 +302,31 @@ namespace pcl
       void
       applyFilter (PCLPointCloud2 &output) override = 0;
   };
+
+
+  template<typename PointT>
+  class FilterIndicesXYZ : public FilterIndices<PointT>
```

Comment:
I don't see the use of this class. Why not simply declare the function of interest in `FilterIndices` and let the derived class choose if they want the default implementation?

[Example](https://godbolt.org/z/fr77zA)

Reply:
Declaring a function in `FilterIndices` that makes use of `xyz` fields will prevent it from ever be instantiated with non-`xyz` point types.

Reply:
This!

Reply:
Didn't notice the access to `xyz` inside. Shouldn't we be filling the `output[ri]` with the `PointT` and not `PointXYZ` anyways?

Reply:
This will change the behavior. Also, what if you don't want to overwrite fields other than `xyz`?